src/input.rs

```diff
@@ -78,6 +79,8 @@ pub struct GamePadState {
 
 pub struct Input {
     pub gamepad_0: GamePadState,
+    // XXX The sdl2 wrapper is quite unsafe: Swapping `controllers` with `sdl` crashes on drop.
```

Comment:
This is probably fixed on more up-to-date versions of the `sdl` crate. I might update SDL after I'm done with upstreaming all of my old patches :)